src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java

```diff
@@ -22,198 +23,164 @@ public class FitnessScoreCalculator implements ScoreCalculator {
+        double fitnessDecrease = calculateFitnessDecrease(contents);
+        double newFitnessScore = fitnessScoreBefore.getValue() - fitnessDecrease;
         int intNewFitnessScore = (int) Math.round(newFitnessScore);
 
+        if (fitnessDecrease == 0) {
+            return fitnessScoreBefore;
+        }
+
         RewardScoreLogEntry logEntry = RewardScoreLogEntry.builder()
                 .date(OffsetDateTime.now())
-                .difference(fitnessScoreBefore.getValue() - intNewFitnessScore)
+                .difference(intNewFitnessScore - fitnessScoreBefore.getValue())
                 .oldValue(fitnessScoreBefore.getValue())
                 .newValue(intNewFitnessScore)
-                .reason(RewardChangeReason.CONTENT_REVIEWED)
-                //.associatedContentIds(Collections.emptyList())
+                .reason(RewardChangeReason.CONTENT_DUE_FOR_REPETITION)
                 .associatedContentIds(getIds(contents))
                 .build();
 
         fitnessScoreBefore.setValue(intNewFitnessScore);
         fitnessScoreBefore.getLog().add(logEntry);
-        fitnessScoreBefore.getLog().clear();
-
-        System.out.println("Fitness Score Before: " + fitnessScoreBefore.getValue());
-        System.out.println("Fitness Score Decrease: " + fitnessDecrease);
-        System.out.println("New Fitness Score: " + newFitnessScore);
 
         return fitnessScoreBefore;
     }
 
     @Override
     public RewardScoreEntity calculateOnContentWorkedOn(AllRewardScoresEntity allRewardScores, List<Content> contents, UserProgressLogEvent event) {
         RewardScoreEntity fitnessScoreBefore = allRewardScores.getFitness();
-        OffsetDateTime today = OffsetDateTime.now();
-        double fitnessDecrease = calculateFitnessDecrease(contents, today);
-        double newFitnessScore = fitnessScoreBefore.getValue() - fitnessDecrease;
-        int intNewFitnessScore = (int) Math.round(newFitnessScore);
 
-        double correctnessBefore = calculateCorrectnessBefore(contents);
+        Content content = getContentOfEvent(contents, event);
+        List<Content> contentsDueForReview = getContentsToRepeat(contents);
 
+        ProgressLogItem latestReview = getLatestReviewExcludingTriggerOfEvent(content);
+        double correctnessBefore = latestReview.getCorrectness();
         double correctnessAfter = event.getCorrectness();
-        int contentsToRepeat = contents.size();
 
-        double fitnessRegen = calculateFitnessRegeneration(newFitnessScore, contentsToRepeat, correctnessBefore, correctnessAfter);
+        double fitnessRegen = calculateFitnessRegeneration(fitnessScoreBefore.getValue(), contentsDueForReview.size(),
+                correctnessBefore, correctnessAfter);
 
-        // Cap the fitness regeneration at 1% per repetition
-        fitnessRegen = Math.min(fitnessRegen, 1);
+        if (fitnessRegen == 0
+            || !event.isSuccess()
+            || wasAlreadyLearnedToday(latestReview)
```

Comment:
Don't create lgo entries when the learning was not successful or it was already learned today.